shared/reducers/chat.js

```diff
@@ -26,19 +26,50 @@ function reducer (state: State = initialState, action: Actions) {
             .set('isLoading', false)
         })
 
-      return state.set('conversationStates', newConversationStates)
+      // Reset the unread count
+      const newInboxStates = state.get('inbox').map(inbox => inbox.get('conversationIDKey') !== conversationIDKey ? inbox : inbox.set('unreadCount', 0))
+
+      return state
+        .set('conversationStates', newConversationStates)
+        .set('inbox', newInboxStates)
     }
     case Constants.appendMessages: {
-      const appendMessages = action.payload.messages
+      const appendAction: AppendMessages = action
+      const appendMessages = appendAction.payload.messages
+      const message: Message = appendMessages[0]
+      const conversationIDKey = appendAction.payload.conversationIDKey
       const newConversationStates = state.get('conversationStates').update(
-        action.payload.conversationIDKey,
+        conversationIDKey,
         initialConversation,
         conversation => conversation.set('messages', conversation.get('messages').push(...appendMessages)))
 
-      return state.set('conversationStates', newConversationStates)
+      const isSelected = state.get('selectedConversation') === action.payload.conversationIDKey
+      let snippet
+      try {
+        // $FlowIssue doens't understand try
+        snippet = makeSnippet(message.message.stringValue(), 100)
+      } catch (_) { }
```

Comment:
so i can remove the try catch but i'd have to replace it with a ton of error checking so it'd go from essentially 1 line to a bunch of null checks. i'll try and lets see what it looks like